src/pykeen/utils.py

```diff
@@ -343,44 +345,131 @@ def save_to_s3(self, directory: str, bucket: str, s3=None) -> None:
+        device: torch.device,
+        initialization: Callable[[nn.Parameter], None] = nn.init.normal_,
+        initialization_kwargs: Optional[Mapping[str, Any]] = None,
+        normalization: Optional[Callable[[torch.FloatTensor], torch.FloatTensor]] = None,
+    ):
+        """Create an embedding object on a device.
+
+        This method is a hotfix for not being able to pass a device during initialization of nn.Embedding. Instead the
+        weight is always initialized on CPU and has to be moved to GPU afterwards.
+
+        :param num: >0
+            The number of embeddings.
+        :param dim: >0
+            The embedding dimensionality.
+        :param device:
+            The device.
+        :param initialization:
+            An optional initializer, which takes a (num_embeddings, embedding_dim) tensor as input, and modifies
+            the weights in-place.
+        :param initialization_kwargs:
+            Additional keyword arguments passed to the initializer
+        :param normalization:
+            A normalization function
+
+        :return:
+            The embedding.
+        """
+        return cls(
+            num=num,
+            dim=dim,
+            initialization=initialization,
+            initialization_kwargs=initialization_kwargs,
+            normalization=normalization,
+        ).to(device=device)
+
+    @property
+    def total_size(self) -> int:  # noqa: D102
+        return self._embeddings.num_embeddings
+
+    @property
+    def dimension(self) -> int:  # noqa: D102
+        return self._embeddings.embedding_dim
+
+    def reset_parameters(self) -> None:  # noqa: D102
+        self.initialization(self._embeddings.weight)
+
+    def forward(self, indices: torch.LongTensor) -> torch.FloatTensor:  # noqa: D102
+        x = self._embeddings(indices)
+        if self.normalization is not None:
+            x = self.normalization(x)
+        return x
+
+
 def get_embedding(
     num_embeddings: int,
     embedding_dim: int,
     device: torch.device,
     initializer_: Optional = None,
     initializer_kwargs: Optional[Mapping[str, Any]] = None,
-) -> nn.Embedding:
-    """Create an embedding object on a device.
-
-    This method is a hotfix for not being able to pass a device during initialization of nn.Embedding. Instead the
-    weight is always initialized on CPU and has to be moved to GPU afterwards.
-
-    :param num_embeddings: >0
-        The number of embeddings.
-    :param embedding_dim: >0
-        The embedding dimensionality.
-    :param device:
-        The device.
-    :param initializer_:
-        An optional initializer, which takes a (num_embeddings, embedding_dim) tensor as input, and modifies the weights
-        in-place.
-    :param initializer_kwargs:
-        Additional keyword arguments passed to the initializer
-
-    :return:
-        The embedding.
-    """
-    # Allocate weight on device
-    weight = torch.empty(num_embeddings, embedding_dim, device=device)
```

Comment:
why don't we have a `weight` in the new implementation?

Reply:
We needed it before, since nn.Embedding initializes on CPU, if we do not explicitly provide a weight. since we now want to intialize on cpu anyway, we can again use the default implementation from torch.